service/resource/cloudformationv2/validate.go

```diff
@@ -0,0 +1,50 @@
+package cloudformationv2
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+)
+
+type validator func(v1alpha1.AWSConfig) error
+
+func (r *Resource) validateCluster(cluster v1alpha1.AWSConfig) error {
+	validators := []validator{
+		r.validateHostPeeringRoutes,
+	}
+
+	for _, v := range validators {
+		if err := v(cluster); err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func (r *Resource) validateHostPeeringRoutes(cluster v1alpha1.AWSConfig) error {
+	input := &ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String("route.destination-cidr-block"),
+				Values: []*string{
+					aws.String(cluster.Spec.AWS.VPC.PrivateSubnetCIDR),
```

Comment:
Nitpick but we could use a key helper as we use the PrivateSubnetCIDR a few times.

Reply:
Good point, done, I've updated all the references to the use the new key helpers.